ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php

```diff
@@ -759,16 +759,17 @@ private function replaceTokens($tokenExpr, $data, $format, $index = NULL) {
    * @param string $key
    * @param mixed $rawValue
    * @param array $data
+   * @param string $dataType = NULL
    * @return array|string
    */
-  protected function formatViewValue($key, $rawValue, $data) {
+  protected function formatViewValue($key, $rawValue, $data, $dataType = NULL) {
```

Comment:
What is the purpose of this new param `$dataType`? I don't see it ever being passed into either this function nor `Download:: formatViewValue`.

Reply:
This is so that `Download::formatViewValue` can pass the result of `$this->getSelectExpression($key)['dataType']` to `AbstractRunAction::formatViewValue`, so we don't have to do it twice for every row we download. Just thought if we're downloading 10k rows, maybe we want to avoid redundant steps.

Reply:
Ok, following the code to `$this->getSelectExpression($key)` it looks pretty efficient, it's just grabbing the value from a cached array. I don't think you're gaining any performance there, but I could see it as a code simplification if you make the `$dataType` param required and do not look it up within this function at all.